build/build.cake

```diff
@@ -291,6 +292,30 @@ Task("AppVeyor")
   .IsDependentOn("Upload-Coverage-Report")
   .IsDependentOn("Publish");
 
+Task("GenerateDocTableOfContents")
+    .Does(() =>
+{
+    var rulesDir = paths.Directories.RootDir.Combine("documentation").Combine("rules");
+    var header = @"
+## Rules
+
+| ID       | Category      | Cause |
+|---|---|---|
+";
+    Information("Generating Table of Contents for {0}", rulesDir);
+    var entries =
+        GetFiles($"{rulesDir}/NS*.md")
+            .Select(f => TableOfContentsEntry.Parse(f))
+            .OrderBy(entry => entry.CheckId);
+    var contents = header + string.Join("\n",
```

Comment:
Replace `\n` with `Environment.NewLine`

Reply:
I wasn't sure about this, as I think the document should stay the same regardless of what platform it is generated on.

Reply:
Fair enough, let's leave it as it is